src/dvclive/live.py

```diff
@@ -241,12 +241,14 @@ def make_report(self):
                 open_file_in_browser(self.report_file)
 
     def end(self):
```

Comment:
Minor: Could you use the `@functools.cache` decorator on `def end(self):` instead of adding `self._ended`?

Reply:
Depends on https://github.com/iterative/dvclive/pull/364#issuecomment-1322159888